src/serializer/nodeSerializers.ts

```diff
@@ -0,0 +1,402 @@
+    }
+    return node.name
+  }
+
+  private serializeAccessElement(node: AccessElementNode): string {
+    return `${this.serializeWithParens(node.base)}[${node.index}]`
+  }
+
+  private serializeArray(node: ArrayNode): string {
+    const elements = node.elements.map((elem) => {
+      let result = this.serializeNode(elem.value)
+      if (elem.isSplat) {
+        result = `...${result}`
+      }
+      return result
+    })
+
+    return `[${elements.join(', ')}]`
+  }
+
+  private serializeArrayCoerce(node: ArrayCoerceNode): string {
+    return `${this.serializeWithParens(node.base)}[]`
+  }
+
+  private serializeObject(node: ObjectNode): string {
+    if (node.attributes.length === 0) {
+      return '{}'
+    }
+
+    this.indent.indent()
+    const attributes = node.attributes.map((attr: ObjectAttributeNode) => {
+      return this.serializeObjectAttribute(attr)
+    })
+    const innerContent = this.indent.newLine() + attributes.join(`,${this.indent.newLine()}`)
+    this.indent.unindent()
+    return `{${innerContent}${this.indent.newLine()}}`
+  }
+
+  private serializeObjectAttribute(attr: ObjectAttributeNode): string {
+    switch (attr.type) {
+      case 'ObjectAttributeValue': {
+        // Check if this is a simple property or a complex expression
+        const simpleKey = this.extractSimplePropertyName(attr.value)
+        if (simpleKey && attr.name === simpleKey) {
+          return this.serializeNode(attr.value)
+        }
+
+        // Special handling for dereferencing projections: author->{...}
+        // Only use shorthand if property name matches the field being dereferenced
+        if (
+          attr.value.type === 'Projection' &&
+          attr.value.base.type === 'Deref' &&
+          attr.value.base.base.type === 'AccessAttribute' &&
+          attr.value.base.base.name === attr.name
+        ) {
+          const space = ' '
+          return `${attr.name}->${space}${this.serializeNode(attr.value.expr)}`
+        }
+
+        // Special handling for simple dereferencing operations
+        // Only use shorthand if property name matches the field being dereferenced
+        if (
+          attr.value.type === 'Deref' &&
+          attr.value.base.type === 'AccessAttribute' &&
+          attr.value.base.name === attr.name
+        ) {
+          return `${attr.name}->`
+        }
+
+        // For all other cases, always use quotes for explicit property names
+        return `"${attr.name}": ${this.serializeNode(attr.value)}`
+      }
+      case 'ObjectSplat':
+        // For object spread, omit 'This' (@) since it's implicit
+        if (attr.value.type === 'This') {
+          return '...'
+        }
+        return `...${this.serializeNode(attr.value)}`
+      case 'ObjectConditionalSplat':
+        return `${this.serializeNode(attr.condition)} => ${this.serializeNode(attr.value)}`
+      default:
+        throw new Error(`Unknown object attribute type: ${(attr as ObjectAttributeNode).type}`)
```

Comment:
Let's use the same trick here
```suggestion
				// @ts-expect-error handle all cases
        throw new Error(`Unknown object attribute type: ${attr.type}`)
```